receiptui/src/test/java/com/hyperwallet/android/ui/receipt/viewmodel/ListUserReceiptViewModelTest.java

```diff
@@ -15,39 +19,51 @@
 @RunWith(RobolectricTestRunner.class)
 public class ListUserReceiptViewModelTest {
 
-    private ReceiptViewModel mReceiptViewModelToTest;
-    private ListUserReceiptViewModel.ListReceiptViewModelFactory mListReceiptViewModelfactoryToTest;
+    private ReceiptViewModel mReceiptViewModel;
+    private ListUserReceiptViewModel.ListReceiptViewModelFactory mListReceiptViewModelFactory;
+    private UserReceiptRepository mUserReceiptRepository;
 
     @Before
     public void initializedViewModel() {
-        UserReceiptRepository userReceiptRepository = new UserReceiptRepositoryImpl();
-        mListReceiptViewModelfactoryToTest = new ListUserReceiptViewModel.ListReceiptViewModelFactory(
-                userReceiptRepository);
-        mReceiptViewModelToTest = mListReceiptViewModelfactoryToTest.create(ReceiptViewModel.class);
+        mUserReceiptRepository = spy(new UserReceiptRepositoryImpl());
+        mListReceiptViewModelFactory = new ListUserReceiptViewModel.ListReceiptViewModelFactory(
+                mUserReceiptRepository);
+        mReceiptViewModel = mListReceiptViewModelFactory.create(ReceiptViewModel.class);
     }
 
     @Test
     public void testIsLoadingData_returnsLiveData() {
-        assertThat(mReceiptViewModelToTest.isLoadingData(), is(notNullValue()));
+        assertThat(mReceiptViewModel.isLoadingData(), is(notNullValue()));
     }
 
     @Test
     public void testGetReceiptErrors_returnsLiveData() {
-        assertThat(mReceiptViewModelToTest.getReceiptErrors(), is(notNullValue()));
+        assertThat(mReceiptViewModel.getReceiptErrors(), is(notNullValue()));
     }
 
     @Test
     public void testGetReceiptList_returnsLiveData() {
-        assertThat(mReceiptViewModelToTest.getReceiptList(), is(notNullValue()));
+        assertThat(mReceiptViewModel.getReceiptList(), is(notNullValue()));
     }
 
     @Test
     public void testGetDetailNavigation_returnsLiveData() {
-        assertThat(mReceiptViewModelToTest.getDetailNavigation(), is(notNullValue()));
+        assertThat(mReceiptViewModel.getDetailNavigation(), is(notNullValue()));
     }
 
     @Test(expected = IllegalArgumentException.class)
     public void testListReceiptViewModelFactory_throwsExceptionOnInvalidClassArgument() {
-        mListReceiptViewModelfactoryToTest.create(ListPrepaidCardReceiptViewModel.class);
+        mListReceiptViewModelFactory.create(ListPrepaidCardReceiptViewModel.class);
+    }
+
+    @Test
+    public void testListUserReceiptViewModel() {
+        verify(mUserReceiptRepository, never()).loadUserReceipts();
+    }
+
+    @Test
+    public void testInit() {
+        mReceiptViewModel.init();
+        verify(mUserReceiptRepository, times(1)).loadUserReceipts();
```

Comment:
verify defaults to execute once no need to specify explicitly

Reply:
Done